ablySpec/RealtimeClient.connection.swift

```diff
@@ -713,6 +713,44 @@ class RealtimeClientConnection: QuickSpec {
                 }
 
             }
+
+            // RTN10
+            context("serial") {
+
+                // RTN10c
+                it("should have last known connection serial from restored connection") {
+                    let client = ARTRealtime(options: AblyTests.commonAppSetup())
+                    defer { client.close() }
+                    let channel = client.channel("test")
+
+                    var lastSerial: Int64 = 0
+                    for _ in 1...5 {
+                        channel.publish("message", cb: { status in
+                            expect(status.state).to(equal(ARTState.Ok))
+                            lastSerial = client.connection().serial
```

Comment:
Correct me if I'm wrong, but there's no guarantee that the last callback called will be that of the last published, since it could take the server more time to respond to any of them. (I've checked the client code and that seems to be indeed what's happening.)